documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -117,27 +108,27 @@ the USM-allocated memory were created for the same queue. For example:
     return 0;
   }
 
+.. note::
+   Use of non-USM pointers is not supported for algorithms with device execution policies.
+
 When using device USM, such as allocated by ``malloc_device``, you are responsible for data
 transfers to and from the device to ensure that input data is device accessible during oneDPL
 algorithm execution and that the result is available to the subsequent operations.
 
-Use std::vector
------------------------------
-
-The following examples demonstrate two ways to use the parallel algorithms with ``std::vector``:
+.. _use-std-vector:
 
-* Host allocators
-* USM allocators
+Use std::vector
```

Comment:
I believe that if a user creates a host-allocated allocator with a std::vector, and then tries to use that data as input to the ranges API via `std::ranges::subrange` for instance, this will not work. I would guess a runtime error, not compile time.  
I don't believe we ever state it does work, but it could be an expectation from a user who is familiar with the iterator based APIs in oneDPL.

Is it worth having a note in this section which explicitly says that host allocated `std::vector` it is unsupported for parallel range algorithms with device execution policies?  Data must be available on the device.

Reply:
Added a note.